cores/esp8266/umm_malloc/umm_malloc_cfgport.h

```diff
@@ -87,5 +93,115 @@ extern char _heap_start[];
 #define UMM_HEAP_STACK_DEPTH 32
 #endif
 
+/*
+ * To support API call, system_show_malloc(), -DUMM_INFO is required.
+ *
+ * UMM_INFO is needed to support several EspClass methods and umm_info().
+ * Partial EspClass method list:
+ *   uint32_t EspClass::getMaxFreeBlockSize()
+ *   void EspClass::getHeapStats(uint32_t* hfree, uint32_t* hmax, uint8_t* hfrag)
+ *   uint8_t EspClass::getHeapFragmentation()
+ *
+ * For the ESP8266 we need an ISR safe function to call for implementing
+ * xPortGetFreeHeapSize(). We can get this with one of these options:
+ *   1) -DUMM_STATS or -DUMM_STATS_FULL
+ *   2) -DUMM_INLINE_METRICS (and implicitly includes -DUMM_INFO)
+ *
+ * If frequent calls are made to ESP.getHeapFragmentation(),
+ * -DUMM_INLINE_METRICS would reduce long periods of interrupts disabled caused
+ * by frequent calls to `umm_info()`. Instead, the computations get distributed
+ * across each malloc, realloc, and free. This appears to require an additional
+ * 116 bytes of IRAM vs using `UMM_STATS` with `UMM_INFO`.
+ *
+ * When both UMM_STATS and UMM_INLINE_METRICS are defined, macros and structures
+ * have been optimized to reduce duplications.
+ *
+ * You can use just UMM_INFO and drop UMM_STATS/UMM_STATS_FULL gaining back
+ * some IROM at the expense of IRAM.
+ *
+ * If you don't require the methods in EspClass that are dependent on UMM_INFO,
+ * you can use just UMM_STATS and save on IROM and a little IRAM.
+ */
+#if defined(UMM_STATS) || defined(UMM_STATS_FULL) || defined(UMM_INLINE_METRICS) || defined(UMM_INFO)
+/*
+  User defined via build options eg. Sketch.ino.globals.h
+*/
+#else
+/*
+  Set expected/implicit defaults for complete support of EspClass methods.
+*/
+#define UMM_INFO 1
+#define UMM_STATS 1
+#endif
+
+/*
+  For `-Dname`, gcc assigns a value of 1 and this works find; however,
+  if `-Dname=0` is used, the intended results will not be obtained.
+
+  Make value and valueless defines compliant with their usage in umm_malloc:
+    `#define name`   => #define name 1
+    `#define name 0` => #undef name
+*/
+#if ((1 - UMM_BEST_FIT - 1) == 2)
```

Comment:
If `UMM_BEST_FIT` were defined with `#define` without a value, it would disappear in the expression and the result would be =2.

I'll update the comments.

Reply:
`1 - - 1` meaning `1 + 1`... ok :)

If umm expects plain flags, we can't just keep using `#defined(FEATURE)` / `#ifdef` / `#ifndef`? But I probably need to re-read the dependencies here